File: liqp-ext/src/main/kotlin/liqp/filters/StripHtml.java
package liqp.filters;

import liqp.nodes.RenderContext;
import org.jsoup.Jsoup;

class StripHtml extends Filter {

    /*
     * strip_html(input)
     *
     * Remove all HTML tags from the string
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        String html = super.asString(value);

        return Jsoup.parse(html).text();
    }
}

File: liqp-ext/src/main/kotlin/liqp/filters/StripHtml.kt
package liqp.filters

import liqp.LValue
import liqp.nodes.RenderContext
import org.jsoup.Jsoup

internal class StripHtml : Filter() {

  /*
     * strip_html(input)
     *
     * Remove all HTML tags from the string
     */
  public override fun apply(context: RenderContext, value: Any?, vararg params: Any): Any {

    val html = LValue.asString(value)

    return Jsoup.parse(html).text()
  }
}
